chain/client/src/sync/state/shard.rs

```diff
@@ -118,7 +113,7 @@ pub(super) async fn run_state_sync_for_shard(
                 );
                 respawn_for_parallelism(&*future_spawner, "state sync download part", future)
             })
-            .buffered(MAX_PARALLELISM_PER_SHARD_FOR_FAIRNESS)
+            .buffered(downloader.task_tracker.limit().saturating_add(1))
```

Comment:
This and the `buffer_unordered` below are only functional changes in this PR.

Reply:
If task tracker will apply rate limiting and assuming we don't care about shard fairness (I ask about in another comment), why do we need buffered at all? Is it just some optimization to avoid too many futures being created at the same time?

Reply:
That's my understanding, yes. And now that this directly queries the task tracker for maximum number of tasks, I don't see a reason to *not* do this.